website/teacher.py

```diff
@@ -253,6 +258,63 @@ def update_level_preferences(user, class_id):
             return {'achievement': achievement}, 200
         return {}, 200
 
+    @app.route('/invite_student', methods=['POST'])
+    @requires_login
+    def invite_student(user):
+        body = request.json
+        # Validations
+        if not isinstance(body, dict):
+            return g.auth_texts.get('ajax_error'), 400
+        if not isinstance(body.get('username'), str):
+            return g.auth_texts.get('username_invalid'), 400
+        if not isinstance(body.get('class_id'), str):
+            return 'class id must be a string', 400
+
+        username = body.get('username')
+        class_id = body.get('class_id')
+
+        if not is_teacher(user):
+            return utils.error_page(error=403, ui_message='retrieve_class')
+        Class = DATABASE.get_class(class_id)
+        if not Class or Class['teacher'] != user['username']:
+            return utils.error_page(error=404, ui_message='no_such_class')
+
+        user = DATABASE.user_by_username(username)
```

Comment:
Nitpicky; invites are case insensitive (you can create an invite for `student` and `Student` even though only `student` exists) - should we ensure that the casing is handled here such that there's no confusion when students create usernames with capitals?

Reply:
username are always lowercase on creation. But it would indeed be best to always parse the username to lowercase.

Reply:
Fixed!